### PR TITLE
Improve reporting for copr builds

### PR DESCRIPTION
**Pull request by dhodovsk**
Fixes https://github.com/packit-service/packit-service/issues/140 and https://github.com/packit-service/packit-service/issues/201 + some stuff along the way

### CODE REVIEW

File: packit_service/constants.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 CONFIG_FILE_NAME = "packit-service.yaml"
 
+PACKIT_ACCOUNT_CHECK = "packit/valid-fas-account"
```
Comment (TomasTomecek) [inline packit_service/constants.py]:
```suggestion
PACKIT_ACCOUNT_CHECK = "packit/authorized-to-build"
```

FAS is Fedora and we want less fedora :)

File: packit_service/worker/copr_build.py [hunk 2]
```diff
@@ -113,25 +112,35 @@ def run_copr_build(self) -> HandlerResults:
...
         if not job.metadata.get("targets"):
             msg = "'targets' value is required in packit config for copr_build job"
-            r.report("failure", msg, check_name=check_name)
+            self.project.pr_comment(self.event.pr_id, msg)
             return HandlerResults(success=False, details={"msg": msg})
 
+        self.job_chroots = job.metadata.get("targets", [])
+        for test_check_name in (f"{PRCheckName.get_testing_farm_check(x)}" for x in self.job_chroots):
+            self.project.set_commit_status(
+                self.event.commit_sha, 'pending', "", "Waiting for a successful rpm build", test_check_name, trim=True
```
Comment (TomasTomecek) [inline packit_service/worker/copr_build.py]:
```suggestion
                self.event.commit_sha, 'pending', "", "Waiting for a successful RPM build", test_check_name, trim=True
```

File: packit_service/worker/fedmsg_handlers.py [hunk 3]
```diff
@@ -66,6 +68,39 @@ def do_we_process_fedmsg_topic(topic: str) -> bool:
     return topic in PROCESSED_FEDMSG_TOPICS
 
 
+def copr_url_from_event(event: CoprBuildEvent):
```
Comment (TomasTomecek) [inline packit_service/worker/fedmsg_handlers.py]: yay, this method will make life easier when working on logs

**Comment by dhodovsk** [comment]
Any ideas why it still fails?

File: packit_service/worker/whitelist.py [hunk 4]
```diff
@@ -195,23 +195,24 @@ def check_and_report(self, event: Optional[Any], project: GitProject) -> bool:
...
+                return False
+            return True
         if isinstance(
             event,
             (CoprBuildEvent, TestingFarmResultsEvent, DistGitEvent, InstallationEvent),
         ):
             return True
+        if isinstance(event, (PullRequestEvent, PullRequestCommentEvent)):
+            account_name = event.github_login
+            assert account_name, f"Failed to get account_name from {type(event)}"
```
Comment (lachmanfrantisek) [inline packit_service/worker/whitelist.py]: `assert` should not be used in the production code -- it was meant for testing/debugging and python code that is run with some optimisations (e.g. `-O` option) can skip the `assert` without executing. 

Can you please fix that also in other occurrences?

**Comment by jpopelka** [comment]
> Any ideas why it still fails?

That's what Rado has been trying to answer since Monday

**Pull request merged**